File: cibcc/Go/step13/cibcc.go
package main

import (
	"os"
	"fmt"
)

func main() {
	if len(os.Args) != 2 {
		fmt.Println("引数の個数が正しくありません")
	}

	token = tokenize([]rune(os.Args[1]))
	prog := program()

	offset := 0
	for va := prog.locals; va != nil; va = va.next {
		offset += 8
		va.offset = offset
	}
	prog.stack_size = offset

	codegen(prog)
}

File: cibcc/c++/step20-22/sect2/cibcc.cpp
#include "cibcc.h"

int main(int argc, char* argv[]) {
	if (argc != 2) {
		std::cout << "引数の個数が正しくありません\n";
	return 1;
	}

	token = tokenize(argv[1]);
	Function *prog = program();
	add_type(prog);

	for (Function *fn = prog; fn; fn = fn->next) {

		int offset = 0;
		for (VarList *vl = fn->locals; vl; vl = vl->next) {
			Var *var = vl->var;
			offset += size_of(var->ty);
			var->offset = offset;
		}
		fn->stack_size = offset;
	}
	codegen(prog);

	return 0;
}

